Stripe/StripeiOSTests/PaymentSheet+APITest.swift

```diff
@@ -209,34 +209,149 @@ class PaymentSheetAPITest: XCTestCase {
     }
 
     func testPaymentSheetLoadWithDeferredIntent() {
-        let expectation = XCTestExpectation(description: "Load PaymentSheet")
+        let loadExpectation = XCTestExpectation(description: "Load PaymentSheet")
+        let confirmExpectation = XCTestExpectation(description: "Confirm deferred intent")
+        let callbackExpectation = XCTestExpectation(description: "Confirm callback invoked")
+
         let types = ["ideal", "card", "bancontact", "sofort"]
         let expected: [STPPaymentMethodType] = [.card, .iDEAL, .bancontact, .sofort]
+        let confirmHandler: PaymentSheet.IntentConfiguration.ConfirmHandler = {_, _ in
+            // TODO(porter) Invoke result callback to finish flow
+            callbackExpectation.fulfill()
+        }
+        let intentConfig = PaymentSheet.IntentConfiguration(mode: .payment(amount: 1000,
+                                                                           currency: "USD",
+                                                                           setupFutureUsage: .onSession),
+                                                            captureMethod: .automatic,
+                                                            paymentMethodTypes: types,
+                                                            confirmHandler: confirmHandler)
+        let elementsSessionJson = STPTestUtils.jsonNamed("ElementsSession")!
+        let elementsSession = STPElementsSession.decodedObject(fromAPIResponse: elementsSessionJson)!
+
+        PaymentSheet.load(
+            mode: .deferredIntent(intentConfig),
+            configuration: self.configuration
+        ) { result in
+            switch result {
+            case .success(let intent, let paymentMethods, _):
+                XCTAssertEqual(
+                    Set(intent.recommendedPaymentMethodTypes),
+                    Set(expected)
+                )
+                XCTAssertEqual(paymentMethods, [])
+                loadExpectation.fulfill()
+
+                let cardParams = STPPaymentMethodCardParams()
+                cardParams.number = "4242424242424242"
+                cardParams.cvc = "123"
+                cardParams.expYear = 32
+                cardParams.expMonth = 12
+                let newCardPaymentOption: PaymentSheet.PaymentOption = .new(
+                    confirmParams: .init(
+                        params: .init(
+                            card: cardParams,
+                            billingDetails: .init(),
+                            metadata: nil
+                        ),
+                        type: .card
+                    )
+                )
```

Comment:
nit: I think this bit is here three times now, can we DRY it

Will make it easier to fix in 2033 when this test starts failing too :P